### PR TITLE
[Stdlib] Change SWIFT_RUNTIME_STDLIB_INTERNAL to not export the symbol.

### PR DESCRIPTION
**Pull request by mikeash**
The functions in LibcShims are used externally, some directly and some through @inlineable functions. These are changed to SWIFT_RUNTIME_STDLIB_SPI to better match their actual usage. Their names are also changed to add "_swift" to the front to match our naming conventions.

Three functions from SwiftObject.mm are changed to SPI and get a _swift prefix.

A few other support functions are also changed to SPI. They already had a prefix and look like they were meant to be SPI anyway. It was just hard to notice any mixup when they were #defined to the same thing.

rdar://problem/35863717

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test

File: stdlib/public/Platform/Platform.swift [hunk 1]
```diff
@@ -176,30 +176,30 @@ public func openat(
...
+  return _swift_stdlib_fcntl(fd, cmd, value)
 }
 
 public func fcntl(
   _ fd: Int32,
   _ cmd: Int32,
   _ ptr: UnsafeMutableRawPointer
 ) -> Int32 {
-  return _stdlib_fcntlPtr(fd, cmd, ptr)
+  return _swift_swift_stdlib_fcntlPtr(fd, cmd, ptr)
```
Comment (gparker42) [inline stdlib/public/Platform/Platform.swift]: This and a few other functions use the prefix `_swift_swift_stdlib_`. They should be `_swift_stdlib_`.

**Comment by gparker42** [comment]
Where do the libc functions get used externally? The entire set of libc shims exists only because libswiftCore can't import the libc's module directly because it would create a circular dependency. Other modules should use libc.

**Comment by jckarter** [comment]
It could be that we end up emitting them because `@inlinable` functions contain references to them. We never want the shim functions themselves to be emitted as ABI entry points, though. Can we mark them always_inline in the header? cc @jrose-apple

**Comment by jckarter** [comment]
Or, if always_inline is inappropriate, could we give them the standard C symbol name with an `asm()` attribute? Would that still cause problems with transitive imports?

**Comment by jrose-apple** [comment]
We can't make them `always_inline` because that implies having an inlinable body in the header, which would defeat the purpose.

`asm` is a little weirder: it looks like we translate `asm` to `_silgen_name`, and I don't trust that we'd get everything correct if a program has two functions with the same `_silgen_name`. If we held off on lowering from the C name to the linkage name until IRGen, instead of in SIL, I'd feel comfortable with it, but as is this seems like a bad idea.

File: stdlib/public/Platform/Platform.swift (hunk 1 above)
Comment (mikeash) [reply stdlib/public/Platform/Platform.swift]: What the heck? The perils of trying to get something for people to look at before closing time on Friday, I guess.

**Comment by compnerd** [comment]
@jckarter  - please do remember to use `__USER_LABEL_PREFIX__` rather than hardcoding it if you go down the `asm` label route.

**Comment by mikeash** [comment]
Many of the shims are dumb wrappers to avoid the circular dependency, but many wrap different underlying OS APIs in one unified function for other stuff to use. For example, `malloc_size` and `random` have different implementations for different targets.

We have a few cross-dylib references within the Swift libraries themselves. `libswiftSwiftPrivate.dylib` uses these:
```
__swift_stdlib_close
__swift_stdlib_read
__swift_stdlib_write
```
And `libswiftDarwin.dylib` uses these:
```
__swift_stdlib_fcntl
__swift_stdlib_getEnviron
__swift_stdlib_getErrno
__swift_stdlib_ioctl
__swift_stdlib_open
__swift_stdlib_setErrno
__swift_swift_stdlib_fcntlPtr
__swift_swift_stdlib_ioctlPtr
__swift_swift_stdlib_openat
```

A few of these functions are being exposed through `@inlineable` calls as well.

`_stdlib_malloc_size` is used in `_ContiguousArrayBuffer.init(_uninitializedCount:minimumCapacity:)`, `ManagedBuffer.capacityInBytes`, and `ManagedBuffer.capacity`.

`_stdlib_strlen` is used in `KeyPath._appendingKeyPaths`.

`_stdlib_random` is used in `SystemRandomNumberGenerator`.

**Comment by jckarter** [comment]
The ones with nontrivial or nonportable logic in them probably shouldn't be in "shims"; that seems misleading. We probably need those entry points in the `Darwin` library to deal with variadic C interfaces, but since Darwin is not the standard library, do the circular dependency issues still matter there? Can we write the Darwin stubs as static inline wrappers around the standard C/POSIX entry points?

**Comment by jrose-apple** [comment]
I think the Darwin ones are holdovers, yes. They can probably be simple wrappers like the other overlay shims.

**Comment by mikeash** [comment]
`random` is the major outlier here, as it has a not-quite-trivial implementation for Windows, and a pretty complex implementation for non-Apple, non-Windows targets. That should probably move to its own file to reduce confusion and keep the "shims" simple. I'll do that and see if I can get rid of the cross-library references from `libswiftDarwin.dylib`.

**Comment by mikeash** [comment]
Coming along. I put `random` in its own file and put all the ones referenced by `libswiftDarwin.dylib` and `libswiftSwiftPrivate.dylib` into a separate header where they live as `static inline` functions. Now I just need to get `malloc_size` and `strlen` taken care of.

**Comment by mikeash** [comment]
@compnerd Can you check over the Windows side of my changes? I tried my best to keep it working but it's very likely I didn't get everything right.

File: stdlib/public/SwiftShims/module.modulemap [hunk 2]
```diff
@@ -19,6 +20,11 @@ module SwiftShims {
   export *
 }
 
+module SwiftShimsInline {
```
Comment (jrose-apple) [inline stdlib/public/SwiftShims/module.modulemap]: Maybe this should be "_SwiftPlatformOverlayShims", to follow the convention below?

**Comment by Azoy** [comment]
Shouldn’t some of these be marked `API`? AFAIK functions like `_swift_stdlib_random` aren’t used in overlay code, but rather `@inlinable` code like the description states. I know `SPI` and `API` both export, but it seems worthwhile marking them `API` if they are and leading with `swift_` prefix.

**Comment by mikeash** [comment]
`random` does seem to fit the definition of `API` from `Visibility.h`.

**Comment by mikeash** [comment]
Changed `random` to be API and rebased on top of the latest `master`. I think this is ready to go, unless there's more feedback. I do want to get this right, so that is welcome. 😄

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Pull request merged**